docs/source/plugin/advanced.rst

```diff
@@ -0,0 +1,67 @@
+.. _plugin-advanced:
+
+======================
+Advanced Tips & Tricks
+======================
+
+Now that you know the basics about plugins, you may have more
+questions, or there is a specific need you want to fulfill but can't quite
+grasp how to implement it. After all, this documentation alone can't cover
+every possible case!
+
+In this chapter, we'll try to share the many tips and tricks both core
+developers and plugin authors have found over time.
+
+If something is not in here, feel free to ask about it on our IRC channel, or
+maybe open an issue with the solution you come up with for your own plugin.
+
+
+Tracking events before/after the bot did
+========================================
+
+When a user joins a channel, or quits the server, Sopel will automatically
+update the information about said user and channel. For example, when they
+join a channel, that information is recorded in
+:attr:`bot.channels <sopel.bot.Sopel.channels>` by adding a new
+:class:`User <sopel.tools.target.User>` object to the correct
+:attr:`channel.users <sopel.tools.target.Channel.users>` dict.
+
+That's all good until you want to do something before or after the change has
+been recorded by Sopel: you need to be careful how you declare your rules.
+
+Before Event
```

Comment:
```suggestion
Before event
```

We seem pretty consistent, at least in the more recent docs, about using `Sentence case` for headings instead of `Title Case`.